common/src/main/java/org/apache/falcon/entity/parser/DatasourceEntityParser.java

```diff
@@ -51,6 +51,10 @@ public DatasourceEntityParser() {
     public void validate(Datasource db) throws FalconException {
         try {
             ClassLoader hdfsClassLoader = HdfsClassLoader.load(db.getName(), db.getDriver().getJars());
+            DatasourceHelper.validateCredential(DatasourceHelper.getCredential(db));
```

Comment:
It might be more readable to make it DatasourceHelper.validateCredentials(...) and internally validate all types.